docs/openapi/lpa-codes-openapi-v1.yml

```diff
@@ -290,34 +306,35 @@ components:
   schemas:
     CodeCreated:
       properties:
-        code:
-          type: string
-        status:
-          type: string
-        id:
-          type: string
-      required:
-        - code
-    CodeRevoked:
-      properties:
-        code:
-          type: string
-        status:
-          type: string
-        id:
-          type: string
+        codes:
+          type: array
+          items:
+            type: object
+            properties:
+              lpa:
+                type: string
+              actor:
+                type: string
+              code:
+                type: string
```

Comment:
I've been setting up [Prism](https://stoplight.io/open-source/prism) in Sirius to provide a mock/stub of this API for local development. The only input is this OpenAPI file, which it uses to generate an HTTP server and create mock responses for each endpoint. It is magic.

Prism has two modes: static and dynamic. Static mode always returns the example data, dynamic mode generates random data based on the specifications. Currently, that means dynamic mode will just return any old string, which is a bit unhelpful. By adding specific formats/patterns, it will match them (more magic!) and give us some realistic data when mocking.

```suggestion
              lpa:
                type: string
                format: uuid
              actor:
                type: string
                format: uuid
              code:
                type: string
                pattern: "[a-zA-Z0-9]{12}"
```

I have [another query](https://github.com/ministryofjustice/opg-data-lpa-codes/pull/6/files#r420802661) about whether we should use UUIDs or Sirius UIDs, which would change this suggestion. But currently it matches the spec as written.

Randomly generated response before:
```json
{"codes":[{"lpa":"eu","actor":"anim consectetur in","code":"ullamco in Excepteur laborum elit"},{"lpa":"officia amet","actor":"labore in","code":"consequat eiusmod"}]}
```

Randomly generated response after:
```json
{"codes":[{"lpa":"c33f5779-0bff-1011-39ff-d4b55acead00","actor":"16a075ce-4692-28f9-f7d3-21a1db65b6ad","code":"WKRdCk9oj5pS"}]}
```

Reply:
I may have over-simplified this slightly, because Prism still doesn't adhere to two rules which are important to us:
- It must return the same number of codes as requested (Prism returns an arbitrary number)
- The lpa/actor UUIDs much match what were originally sent (Prism generates random UUIDs)

I think the suggested change is still useful, but we might not be able to use Prism as seamlessly as I hoped.

Reply:
I will leave them as unformatted strings for now, we will revisit the format later if required. I wanted it to remain flexible for UUIDs, random strings, db ids...

Prism can do a lot of things, but if you want to do specific stuff you might have to write some Python.